### PR TITLE
Module publish

### PR DESCRIPTION
**Pull request by tikhono**

### CODE REVIEW

File: pallets/sp-mvm/src/lib.rs [hunk 1]
```diff
@@ -220,17 +224,37 @@ pub mod pallet {
             module_bc: Vec<u8>,
             gas_limit: u64,
         ) -> DispatchResultWithPostInfo {
-            let who = ensure_signed(origin)?;
-            debug!("executing `publish` with signed {:?}", who);
+            // Allows to update Standard Library if root.
+            let (sender, signer) = match T::UpdaterOrigin::ensure_origin(origin.clone()) {
+                Ok(_) => {
+                    debug!("executing `publish module` with root");
+                    (CORE_CODE_ADDRESS, sudo::Pallet::<T>::key())
```
Comment (borispovod) [inline pallets/sp-mvm/src/lib.rs:231]: Remove sudo, you really don't need it and it's not correct way.
Instead use: `Ok(signer)` instead of `Ok(_)`

**Review by borispovod — CHANGES_REQUESTED** [review]

File: pallets/sp-mvm/src/lib.rs [hunk 2]
```diff
@@ -220,17 +224,37 @@ pub mod pallet {
...
+            let vm = Self::get_vm()?;
+            let gas = Self::get_move_gas_limit(gas_limit)?;
+
+            let tx = {
+                let sender = addr::account_to_bytes(&sender);
+                debug!("converted sender: {:?}", sender);
+                ModuleTx::new(module_bc, AccountAddress::new(sender))
+            };
+
+            let vm_result = vm.publish_module(gas, tx, false);
```
Comment (fzzr-) [inline pallets/sp-mvm/src/lib.rs:249]: Would be great use `raw_publish_module` here and so move this root-determining and address conversion into the `raw_publish_module`.
Also you can make `raw_publish_module` accept `origin: OriginFor<T>` if needed.

File: pallets/sp-mvm/src/lib.rs [hunk 3]
```diff
@@ -220,17 +220,39 @@ pub mod pallet {
...
```
Comment (fzzr-) [inline pallets/sp-mvm/src/lib.rs:249]: Would be great continue to use `raw_publish_module` here and so move VM call into the raw_publish_module.
You can make raw_publish_module accept origin: OriginFor<T> if needed.

Comment (fzzr-) [reply pallets/sp-mvm/src/lib.rs:249]: _Also, __maybe__ will better to move there part about origin-determining and address conversion (lines 223-237), but I'm not sure._

**Review by fzzr- — CHANGES_REQUESTED** [review]
Almost great!

File: pallets/sp-mvm/src/lib.rs (hunk 2 above)
Comment (fzzr-) [reply pallets/sp-mvm/src/lib.rs:249]: Or just define special method for it! And name it `ensure_and_convert`.
_This is the same algorithm as in [`publish_package` method](https://github.com/pontem-network/pontem/pull/170/files#diff-4e5930d2e2ba9c4d86fb707cc5d1eb7aef9839019ea8e6f3125310aa710557cbL255-R287).

File: pallets/sp-mvm/src/lib.rs (hunk 3 above)
Comment (fzzr-) [reply pallets/sp-mvm/src/lib.rs:249]: Or just define special method for it! And name it `ensure_and_convert`.
_This is the same algorithm as in [`publish_package` method](https://github.com/pontem-network/pontem/pull/170/files#diff-4e5930d2e2ba9c4d86fb707cc5d1eb7aef9839019ea8e6f3125310aa710557cbL255-R287).

Comment (tikhono) [reply pallets/sp-mvm/src/lib.rs]: I reused  `raw_publish_module`

File: Cargo.toml [hunk 4]
```diff
@@ -10,6 +10,8 @@ members = [
 [profile.release]
 panic = 'unwind'
 rpath = false
+debug = true
```
[the remark this answers is not in the document]
Comment (tikhono) [reply Cargo.toml]: Comment this so there will be an easy way to enable debug, but now it is disabled.

**Review by fzzr- — APPROVED** [review]

**Review by borispovod — APPROVED** [review]

**Pull request merged**